### DISCUSSION TITLE
How do I scroll the function description in the autocomplete prompt window?

### DISCUSSION THREAD

**Discussion by rishitc** [helix-editor/helix#6647] [Q&A] [answered]
Hi everyone,
I'm facing an issue where I cannot scroll the documentation for the highlighted option in the autocomplete menu.
Often, the documentation of an autocomplete option gets cut off when it is too long, and there are few lines remaining between the line I'm coding on and the end of the screen. 

Initially, I thought there would be an option to scroll the documentation of the highlight option in the autocomplete menu, but `Ctrl-u` and `Ctrl-d` don't work here. (I tried these keys because of [this](https://docs.helix-editor.com/keymap.html#popup))
These key bindings only work when I view the documentation of an API when I open it using the `<space> + k` option ([ref](https://docs.helix-editor.com/keymap.html#space-mode:~:text=Show%20documentation%20for%20item%20under%20cursor%20in%20a%20popup%20(LSP)))

Can I scroll the documentation of the highlighted option when browsing the different autocomplete options in Helix? If yes, what is the right way to do it?

I've added an image highlighting the (incompletely) rendered documentation for the selected autocomplete option, which I cannot scroll. Due to this, I need help reading the complete documentation of an autocomplete option.

![Helix doc scroll issue](https://user-images.githubusercontent.com/12199011/230651891-86f25160-6422-4492-9a26-e95daa0f7a7d.png)

Here are a few details about my workflow:
1. Programming Langauge used?
    -  Rust
1. Any issues noticed in `hx --health` for Rust?
    - No
1. Terminal Emulator used?
    - iTerm2 V3.4.19

**Comment by the-mikedavis** [comment, marked as answer]
There is an issue for this: https://github.com/helix-editor/helix/issues/3062

**Reply by rishitc** [reply]
Hi @the-mikedavis,
Thank you so much for linking the issue. It describes the same issue I'm facing now 👍

**Comment by rishitc** [comment]
I have a follow-up question: In the linked issue, there is a PR which links back to it, that claims to add the scroll behaviour for auto-complete pop-ups. However, it was never merged (I'm referring to PR #4364).

I would love to help in getting this feature merged, but other than the "scroll-down hint feature", I was not able to find any other issues with the existing PR (at least from its discussion section). Is that the blocker for this PR?

**Reply by rishitc** [reply]
Edit: I now see that the "scroll-down hint feature" was covered in a separate (now merged) PR.

**Reply by rishitc** [reply]
I'll try resolving the merge conflicts in the current PR #4364 and try submitting a new PR for this feature 👍
